unfurl/src/main/kotlin/me/saket/unfurl/extension/HtmlTagsBasedUnfurler.kt

```diff
@@ -33,8 +34,8 @@ open class HtmlTagsBasedUnfurler : UnfurlerExtension {
 
         if (body != null && body.contentType().isHtmlText()) {
           // TODO: stream the HTML body only until a "</head>" is received instead of streaming the entire HTML body.
-          Jsoup.parse(
-            /* in */ body.source().inputStream(),
+          Ksoup.parse(
+            /* in */ BufferReader(body.bytes()),
```

Comment:
Does `ksoup` have a streaming API? Calling `bytes()` is dangerous here as it'll allocate the entire HTML in memory.

Reply:
Have switched this to buffer. Can you take a look?